File: Lab/SEM4/MPP/lab9/moto_c/src/Common/sync/Request.cs
namespace moto_c.Common.sync;

public class Request
{
    public RequestType type;
    public string message;

    public Request(RequestType type, string message)
    {
        this.type = type;
        this.message = message;
    }

    public RequestType GetType()
    {
        return type;
    }

    public string GetMessage()
    {
        return message;
    }

    public override string ToString()
    {
        return type + "|" + message;
    }

    public static Request FromString(string str)
    {
        string[] parts = str.Split('|', 2);
        if (parts.Length != 2)
            throw new ArgumentException("Invalid request: " + str);

        return new Request(
            Enum.Parse<RequestType>(parts[0]),
            parts[1]
        );
    }
}

File: Lab/SEM4/MPP/lab13/moto_java/common/src/main/java/org/zdroba/sync/Request.java
package org.zdroba.sync;

import java.io.Serializable;

public class Request implements Serializable {
    private RequestType type;
    private String message;

    public Request(RequestType type, String message) {
        this.type = type;
        this.message = message;
    }

    public RequestType getType() {
        return type;
    }

    public String getMessage() {
        return message;
    }

    @Override
    public String toString() {
        return type + "|" + message;
    }

    public static Request fromString(String str) {
        String[] parts = str.split("\\|", 2);
        if (parts.length != 2) throw new IllegalArgumentException("Invalid request: " + str);
        return new Request(RequestType.valueOf(parts[0]), parts[1]);
    }
}